### PR TITLE
Allows nuke ops to purchase reinforcements

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
For 25 telecrystals you get another team member. He spawns in the basic gear (turtleneck, stetchkin, ID, implants, etc) but you'll have to provide them with a suit/gun, or other way on station. 

:cl: Kor
rscadd: Nuke ops can now purchase additional team members for 25 telecrystals. They don't come with any gear other than a pistol however, so remember to save some points for them.
/:cl:

### CODE REVIEW

**Comment by Cebasiz** [comment]
this is a really good idea, though no idea in practice if 25tc is low/high enough.

real downfall of nukeops is action economy, hands, etc, not weapons or toughness. 

having 6-7 guys means you can split up and do a blitzkrieg on two different locations, causing mass confusion and allowing you to split the attention of security, or enables you to just have a fuckload more guys in one place.

it also i guess would heighten your chances of having competent nukes on any one team, considering you'll have more dudes.

basically it's cool

**Comment by KorPhaeron** [comment]
>though no idea in practice if 25tc is low/high enough.

Me neither, especially considering you gotta arm them afterwards.

**Comment by Thunder12345** [comment]
Breaks the "We killed five we're done here" meta, too

**Comment by Cebasiz** [comment]
25+8(suit)+12(minimum?) seems too much. you might run into a problem getting anyone to buy it, with maybe the exception of those planning to use a teleporter board.

you don't have to buy a suit if you're teleporting, and the more people you plan on teleporting in the less it costs comparatively (don't have to buy suits)

if you're going to add this kor, i'd consider adding more pinpointers/flashbangs/tear gas to the shuttle, as an op without a pinpointer is basically SOL.

also coderbus should really consider lowering the costs of nukeop weaponry to make this more attractive.

**Comment by KorPhaeron** [comment]
@Cebasiz 

They get one free extra suit on the shuttle.

You could also do stuff like crack the armory and arm you guys that way.

Or arm them in challenge ops when you're swimming in points anyway.

Probably a bit too expensive but I'd rather err on the side of it being underpowered than overpowered

**Comment by WJohn** [comment]
Very cool. :+1: 

Makes good use of the new 200 TC late nuke ops tactic.

**Comment by Cebasiz** [comment]
ah, okay. i'd still add more pinpointers though as i don't think you can buy them(?)

as a challenge mode thing it'll probably be necessary, as 5 guys probably won't have a chance against a prepared crew. in fact, i'd probably take 8 minimum and use a teleporter or something.

tl;dr we'll see. it adds options and we really need those in nuke.

**Comment by KorPhaeron** [comment]
@Wjohn

It is actually 280 now.

Which is exactly enough for 10+the drop pod if you feel like a stetchkin rush

**Comment by LatD** [comment]
I'll be waiting for a round where the ops spend all their TCs on more people and go in stealth

**Comment by Incoming5643** [comment]
:-1: Another "I better observe this round for the chance at free antag" role. It's a good idea, but reinforces bad game design.

Things like apprentices, syndiborgs, and this should problably somehow draw from living players, and not the dead.

**Comment by xxalpha** [comment]
25TC seems a good price to start with until you know how well this works.

File: code/game/gamemodes/antag_spawner.dm [hunk 1]
```diff
@@ -135,35 +150,59 @@
...
+			var/mob/living/carbon/human/M = new/mob/living/carbon/human(T)
+			C.prefs.copy_to(M)
+			M.key = C.key
+			var/obj/machinery/nuclearbomb/nuke = locate("syndienuke") in nuke_list
+			if(nuke)
+				nuke.r_code = nuke_code
+			M.mind.make_Nuke(T, nuke_code, 0, FALSE)
+
+
+/obj/item/weapon/antag_spawner/borg_tele/operative
```
Comment (KazeEspada) [inline code/game/gamemodes/antag_spawner.dm]: Why not it's own type path from anag_spawner?

Comment (KorPhaeron) [inline code/game/gamemodes/antag_spawner.dm]: Because it'd be a massive amount of copypaste with all the checks/messages and then someone would yell at me

**Comment by Robustin** [comment]
25 seems fine for not getting a viable weapon or suit. 

Giving them a good weapon, nukeop suit, ammo, etc. would end up costing around what a syndieborg costs and we know how syndieborgs go. 

Potential problems: 

1) Rush armory (i.e. emagging through secpod airlock or shooting into locker room), Poop out like 5 more dudes, they arm themselves with the armory, Nuke Ops becomes Nuke Army 

2) People getting reinforcements and forgetting they need to provide gear for them, leaving them with shitty ops stuck on the shuttle. 

Also lol @Incoming5643 

"WOULD YOU LIKE TO BE A NUKEOP REINFORCEMENT?: YES / NO"
*Clicks No*
"Uh guys, did we secure the disk?"

**Comment by KorPhaeron** [comment]
> Poop out like 5 more dudes

I'll make the teleporter unusable out of the base to prevent this

**Comment by Iamgoofball** [comment]
@Incoming5643 Unfortunately, pulling from live players causes a few problems.

1. If players start randomly going braindead, and their ckey is still connected in Who, people will start screaming to check for new antags.
2. above mentioned thing by robustin

Pulling from dead players prevents this, and helps keep people entertained while dead and still connected to the server, because they'll be watching the new antags, and hoping they roll it themselves.

**Comment by Incoming5643** [comment]
You don't say "WOULD YOU LIKE TO BE A NUKEOP REINFORCEMENT?: YES / NO", you just automate the nonspecific "need a prayer" request to people who are down for midround antag.

Which is a setting people already have by the way.

I don't know how to really solve the "scanning who for people who look DC'd but aren't" bit, but that's really shitty behavior and leads to having people cut out really basic features like when we had to remove the list/number of players ready up'd at roundstart because people would check it against the crew list to see how many offstation antags there were.

**Comment by Zilenan91** [comment]
We really do need to lower Nuke weapon costs though, they're way way too much as-is.

**Comment by Shapsy** [comment]
Prevent people who suicide from getting selected, isn't that how golems work or something? Besides, in all honesty the admin team should probably be cracking down on people who suicide at roundstart all the time anyway

**Comment by Lumipharon** [comment]
Neat idea.

**Comment by PKPenguin321** [comment]
are these refundable if no ghosts are available, like syndieborgs?

**Comment by KorPhaeron** [comment]
@PKPenguin321 

Yes

**Comment by PKPenguin321** [comment]
tite
:+1:

**Comment by Zilenan91** [comment]
@KorPhaeron Can you make it so the syndies that get spawned in are wearing an armor vest or would that be too much value to handle.

File: code/game/gamemodes/antag_spawner.dm [hunk 2]
```diff
@@ -127,6 +141,7 @@
 	var/TC_cost = 0
 	var/borg_to_spawn
 	var/list/possible_types = list("Assault", "Medical")
+	var/spawn_type = BORG_TELE
```
Comment (tkdrg) [inline code/game/gamemodes/antag_spawner.dm]: Instead of making a magical define and switch on that, you could just move this code down from `/obj/item/weapon/antag_spawner/borg_tele` to, say, `/obj/item/weapon/antag_spawner/nukeops/borg_tele`. Then, in the places where you checked for `BORG_TELE`, you could call a proc (for example, `select_type()` or `spawn_antag()` and override its behavior on the children `/obj/item/weapon/antag_spawner/nukeops/borg_tele` and `/obj/item/weapon/antag_spawner/nukeops/ops_tele`.

Comment (KorPhaeron) [inline code/game/gamemodes/antag_spawner.dm]: I didnt want to change the borg tele path because I am unsure what that would do to stat tracking.

Comment (tkdrg) [inline code/game/gamemodes/antag_spawner.dm]: @MrStonedOne could you chime in here?

Comment (KorPhaeron) [inline code/game/gamemodes/antag_spawner.dm]: I mean I guess it'd only kill borg tracking for a month at worst

Comment (MrStonedOne) [inline code/game/gamemodes/antag_spawner.dm]: Ya, it would just split the vote so to speak

Comment (KorPhaeron) [inline code/game/gamemodes/antag_spawner.dm]: Well I've  got both versions ready to go @MrStonedOne @tkdrg  so just let me know which you prefer

Comment (tkdrg) [inline code/game/gamemodes/antag_spawner.dm]: I've already stated my opinion, I think OOP is better than huge switches.

Comment (KorPhaeron) [inline code/game/gamemodes/antag_spawner.dm]: Separation between the server and erro's stats page is absolute then

File: code/datums/uplink_item.dm [hunk 3]
```diff
@@ -248,17 +248,26 @@ var/list/uplink_items = list()
...
-	item = /obj/item/weapon/antag_spawner/borg_tele
+	item = /obj/item/weapon/antag_spawner/nuke_ops/borg_tele
 	cost = 50
 	gamemodes = list(/datum/game_mode/nuclear)
 	surplus = 0
 
 //for refunding the syndieborg teleporter
 /datum/uplink_item/dangerous/syndieborg/spawn_item()
-	var/obj/item/weapon/antag_spawner/borg_tele/T = ..()
+	var/obj/item/weapon/antag_spawner/nuke_ops/T = ..()
```
Comment (tkdrg) [inline code/datums/uplink_item.dm]: `nuke_ops/borg_tele`

File: code/datums/uplink_item.dm [hunk 4]
```diff
@@ -248,17 +248,26 @@ var/list/uplink_items = list()
...
 
+
+/datum/uplink_item/dangerous/reinforcement
+	name = "Reinforcements"
+	desc = "Call in an additional team member. They won't come with any gear, so you'll have to save some telecrystals to arm them as well."
+	item = /obj/item/weapon/antag_spawner/nuke_ops
+	cost = 25
+	gamemodes = list(/datum/game_mode/nuclear)
+	surplus = 0
+
```
Comment (tkdrg) [inline code/datums/uplink_item.dm]: Should probably do the same `TC_cost` shebang as above. Might make sense to make it a child.

File: code/game/gamemodes/antag_spawner.dm [hunk 5]
```diff
@@ -119,40 +122,81 @@
...
-	desc = "A single-use teleporter designed to deploy a single Syndicate cyborg onto the field."
+
+
+
+
+
+///////////BORGS AND OPERATIVES
+
+
+/obj/item/weapon/antag_spawner/nuke_ops
```
Comment (tkdrg) [inline code/game/gamemodes/antag_spawner.dm]: This works I guess, not exactly what I had in mind (I expected two children of this, one for nukies and one for borgies), but this is acceptable.

File: code/datums/uplink_item.dm (hunk 4 above)
Comment (KorPhaeron) [inline code/datums/uplink_item.dm]: datum/uplink_item/dangerous/reinforcement/borg_tele and /nuke_op?

Comment (tkdrg) [inline code/datums/uplink_item.dm]: makes sense

**Comment by KorPhaeron** [comment]
@tkdrg 

All done for real now, waiting on travis
